src/python/pants/option/options_integration_test.py

```diff
@@ -108,3 +108,32 @@ def test_pants_symlink_workdirs(tmp_path: Path) -> None:
     pants_run.assert_success()
     # Make sure symlink workdir is pointing to physical workdir
     assert Path(os.readlink(symlink_workdir.as_posix())) == physical_workdir
+
+
+def test_fromfile_invalidation(tmp_path: Path) -> None:
+    workdir = (tmp_path / "workdir").as_posix()
+
+    fromfile_path = tmp_path / "fromfile.txt"
+    fromfile_path.write_text("dist1")
+    pants_run = run_pants_with_workdir(
+        [f"--pants-distdir=@{fromfile_path}"],
+        use_pantsd=True,
+        workdir=workdir,
+    )
+    assert "Scheduler initialized." in pants_run.stderr
+
+    pants_run = run_pants_with_workdir(
+        [f"--pants-distdir=@{fromfile_path}"],
+        use_pantsd=True,
+        workdir=workdir,
+    )
+    assert "Scheduler initialized." not in pants_run.stderr
```

Comment:
Would it be better to verify that the same pantsd process is still running?

Reply:
It kinda does because the "Initialization options changed: reinitializing scheduler..." message is only printed by a running pantsd that is recreating its scheduler, but I can make this more explicit, since it also logs its pid.

Reply:
PTAL